lieer/gmailieer.py

```diff
@@ -617,25 +691,57 @@ def set (self, args):
     if args.timeout is not None:
       self.local.state.set_timeout (args.timeout)
 
-    if args.replace_slash_with_dot:
-      self.local.state.set_replace_slash_with_dot (args.replace_slash_with_dot)
-
-    if args.no_replace_slash_with_dot:
-      self.local.state.set_replace_slash_with_dot (not args.no_replace_slash_with_dot)
-
     if args.drop_non_existing_labels:
       self.local.state.set_drop_non_existing_label (args.drop_non_existing_labels)
 
     if args.no_drop_non_existing_labels:
       self.local.state.set_drop_non_existing_label (not args.no_drop_non_existing_labels)
 
+    new_label_translation_value = None
+    # the following two settings are mutual exclusive. if none of them
+    # is True, leave the state as it is.
+    if args.user_label_translation:
+      new_label_translation_value = True
+    elif args.no_user_label_translation:
+      new_label_translation_value = False
+
+    # if requested to change label translation setting, and the new state
+    # is different from the current state, make the change
+    if new_label_translation_value is not None \
+       and new_label_translation_value != self.local.state.user_label_translation:
+      self.confirm_and_set_label_translation(new_label_translation_value)
+
     print ("Repository info:")
     print ("Account ...........: %s" % self.local.state.account)
     print ("Timeout ...........: %f" % self.local.state.timeout)
     print ("historyId .........: %d" % self.local.state.last_historyId)
     print ("lastmod ...........: %d" % self.local.state.lastmod)
     print ("drop non labels ...:", self.local.state.drop_non_existing_label)
-    print ("replace . with / ..:", self.local.state.replace_slash_with_dot)
+    print ("Use user's label translation: {}".format(
+      self.local.state.user_label_translation))
 
 
 
+  def confirm_and_set_label_translation(self, new_label_translation_value):
+    """Set the state variable of label translation to new value"""
+
+    msg = "You asked to change the user label translation setting\n" \
+          + "from '{}' to '{}'. Changing this setting might \n" \
+          + "change the way message labels are translated between your\n" \
+          + "GMail account and your Notmuch message tags."
+
+    question = "Are you sure you want to change this setting?"
+
+    print(msg.format(self.local.state.user_label_translation,
+                     new_label_translation_value))
+
+    reply = str(input(question + ' [y/N]: ')).lower().strip()
+
+    if reply[:1] == "y":
+      self.local.state.set_user_label_translation(new_label_translation_value)
+      print(">> User label translation {}".format(
+        ['DISABLED', 'ENABLED'][new_label_translation_value]))
+    else:
+      print(">> User label translation setting not changed")
+
+    print("")
```

Comment:
I would like to keep interactiveness at a minimum (we have to be interactive with getting oauth tokens). I think it would be better to use something like:

```
* gmi set --user-label-translation file -> only disaplays warning
* gmi set --user-label-translation --yes file actual sets option
```

Or we could just have no warning on setting but list it in (partially) big types in the help output / readme.

Reply:
just to make sure - 'gmi set --user-label-translation' sets a boolean flag in the state file. 
Just to verify I understand your suggestion, do you mean that without the 'force' switch we should issue a warning similar to the current one, and without the 'force' switch we should go ahead and set the flag without warnings?
If this is it, I think it would be a good solution.